pkg/syncer/metadatasyncer.go

```diff
@@ -2909,3 +2915,80 @@ func startStoragePolicyQuotaCRInformer(ctx context.Context, cfg *restclient.Conf
+	}
+	// Prepare Config and NewClientForGroup for cnsOperatorClient
+	restConfig, err := config.GetConfig()
+	if err != nil {
+		log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+		return
+	}
+	cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+	if err != nil {
+		log.Errorf("Failed to create CnsOperator client. Err: %+v", err)
+		return
+	}
+	// Get the list of all StoragePolicyUsage CRs from all supervisor namespaces
+	storagePolicyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+	err = cnsOperatorClient.List(ctx, storagePolicyUsageList)
+	if err != nil {
+		log.Errorf("failed to list %s CR from all supervisor namespaces. Error: %+v",
+			storagepolicyusagev1alpha1.CRDSingular, err)
+		return
+	}
+	// Get the list of all CnsVolumeInfo CRs from vmware-system-csi namespace
+	namespace := common.GetCSINamespace()
+	cnsVolumeInfoList := &cnsvolumeinfov1alpha1.CNSVolumeInfoList{}
+	err = cnsOperatorClient.List(ctx, cnsVolumeInfoList, client.InNamespace(namespace))
+	if err != nil {
+		log.Errorf("failed to list %s CR from supervisor namespace %q. Error: %+v",
+			cnsvolumeinfov1alpha1.CRDSingular, namespace, err)
+		return
+	}
+	cnsVolumeInfoMap := make(map[string]*cnsvolumeinfov1alpha1.CNSVolumeInfo)
+	for _, volumeInfo := range cnsVolumeInfoList.Items {
+		cnsVolumeInfoMap[volumeInfo.Name] = volumeInfo.DeepCopy()
+	}
+	if len(cnsVolumeInfoList.Items) > 0 {
+		// Iterate through storagePolicyUsageList
+		for _, storagePolicyUsage := range storagePolicyUsageList.Items {
+			totalUsedQty := resource.NewQuantity(int64(0), resource.BinarySI)
+			// For every storagePolicyUsage, fetch "Bound" PVs in that namespace from k8sVolumesToNamespaceMap
+			if volumes, ok := k8sVolumesToNamespaceMap[storagePolicyUsage.Namespace]; ok {
+				for _, pv := range volumes {
+					// Verify the StorageClass, StoragePolicyId match with the storagePolicyUsage spec
+					// using the CnvOlumeInfo CR for the volume
+					if cnsVolumeInfo, ok := cnsVolumeInfoMap[pv.Spec.CSI.VolumeHandle]; ok {
+						if cnsVolumeInfo.Spec.StorageClassName == storagePolicyUsage.Spec.StorageClassName &&
+							cnsVolumeInfo.Spec.StoragePolicyID == storagePolicyUsage.Spec.StoragePolicyId &&
+							cnsVolumeInfo.Spec.Namespace == storagePolicyUsage.Namespace {
+							totalUsedQty.Add(*resource.NewQuantity(cnsVolumeInfo.Spec.Capacity.Value(), cnsVolumeInfo.Spec.Capacity.Format))
+						}
+						patchedStoragePolicyUsage := *storagePolicyUsage.DeepCopy()
+						patchedStoragePolicyUsage.Status.ResourceTypeLevelQuotaUsage.Used = totalUsedQty
+						err := patchStoragePolicyUsage(ctx, cnsOperatorClient, &storagePolicyUsage, &patchedStoragePolicyUsage)
+						if err != nil {
+							log.Errorf("Patching operation failed for StoragePolicyUsage CR: %q in namespace: %q. err: %v",
+								storagePolicyUsage.Name, patchedStoragePolicyUsage.Namespace, err)
+							return
+						}
+					}
```

Comment:
Can we add a Info log here to print "ResourceTypeLevelQuotaUsage.Used" has been updated to value "totalUsedQty"?

Reply:
Done